pipeline/detection.py

```diff
@@ -110,10 +113,40 @@ def run(self, *args, **kwargs):
         return ds
 
     def extract_sources(self, image):
+        """
+        Run source-extraction (using SExtractor) on the given image.
+
+        Parameters
+        ----------
+        image: Image
+            The image to extract sources from.
+
+        Returns
+        -------
+        sources: SourceList
+            The list of sources detected in the image.
+            This contains a table where each row represents
+            one source that was detected, along with all its properties.
+
+        """
         # TODO: finish this
-        # TODO: this should also generate an estimate of the PSF
+        # TODO: this should also generate an estimate of the PSF?
```

Comment:
Thought needed.  The reason not to do it here is that you aren't going to want to generate a PSF estimate when detecting sources on a subtraction image.  So, we should separate source extraction and PSF estimation (with the latter requiring the former as a prerequisite).

Reply:
The problem is that you need stars to find a PSF, and then you want to PSF to find stars in a more sensitive way (using a matched filter with the correct PSF). It would maybe make sense to make PSF finding an option to the sextractor run.

Reply:
Yeah... so there are different modes of operation.

In fact, it's more involved... I'll make an issue.  Let's just leave this comment as is for now.